recipes/aravis/all/conanfile.py

```diff
@@ -58,62 +53,81 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        self.options["glib"].shared = True
-
-    def validate(self):
-        if is_msvc(self) and self.settings.get_safe("compiler.runtime", "").startswith("MT"):
-            raise ConanInvalidConfiguration("Static MT/MTd runtime is not supported on Windows due to GLib issues")
-        if not self.options["glib"].shared and self.options.shared:
-            raise ConanInvalidConfiguration("Shared Aravis cannot link to static GLib")
-        if self.settings.os == "Macos":
-            raise ConanInvalidConfiguration("macOS builds are disabled until conan-io/conan#7324 gets merged to fix macOS SIP issue #8443")
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+        if self.options.shared:
+            self.options["glib"].shared = True
 
-    def build_requirements(self):
-        self.build_requires("meson/0.63.3")
-        self.build_requires("pkgconf/1.9.3")
-        if self.options.introspection:
-            self.build_requires("gobject-introspection/1.72.0")
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("glib/2.74.0")
-        self.requires("libxml2/2.9.14")
-        self.requires("zlib/1.2.12")
+        self.requires("glib/2.75.2")
+        self.requires("libxml2/2.10.3")
+        self.requires("zlib/1.2.13")
         if self.options.usb:
             self.requires("libusb/1.0.26")
         if self.options.gst_plugin:
             self.requires("gstreamer/1.19.2")
             self.requires("gst-plugins-base/1.19.2")
 
-    def export_sources(self):
-        export_conandata_patches(self)
+    def package_id(self):
+        self.info.requires["glib"].full_package_mode()
+        if self.options.gst_plugin:
+            self.info.requires["gstreamer"].full_package_mode()
+            self.info.requires["gst-plugins-base"].full_package_mode()
+
```

Comment:
```suggestion
```
I recommend excluding full package mode here, otherwise, it may fail with missing packages as historically we have on CCI.

Reply:
It was already there and I don't know the consequences for those who decided to add this :s

Reply:
I know it was there before. But I still prefer risking it than keep missing packages appearing on CCI. Anyway, we have an issue listed about it: https://github.com/conan-io/conan-center-index/issues/11684